pyfiglet/__init__.py

```diff
@@ -169,6 +182,38 @@ def infoFont(cls, font, short=False):
                 infos.append(line)
         return '\n'.join(infos) if not short else infos[0]
 
+    @staticmethod
+    def installFonts(file_name):
+        """
+        Install the specified font file to this system.
+        """
+        if isinstance(pkg_resources.get_provider('pyfiglet'), pkg_resources.ZipProvider):
+            # Figlet is installed using a zipped resource - don't try to upload to it.
+            location = SHARED_DIRECTORY
+        else:
+            # Figlet looks like a standard directory - so lets use that to install new fonts.
+            location = pkg_resources.resource_filename('pyfiglet', 'fonts')
+
+        print("Installing {} to {}".format(file_name, location))
+
+        # Make sure the required destination directory exists
+        if not os.path.exists(location):
+            os.makedirs(location)
+
+        # Copy the font definitions - unpacking any zip files as needed.
+        if os.path.splitext(file_name)[1].lower() == ".zip":
+            # Ignore any structure inside the ZIP file.
+            with zipfile.ZipFile(file_name) as zip_file:
+                for font in zip_file.namelist():
+                    font_file = os.path.basename(font)
+                    if not font_file:
+                        continue
+                    src = zip_file.open(font)
+                    dest = open(os.path.join(location, font_file), "wb")
```

Comment:
This looks like we're missing  closes on these two opens? Can they be moved into a `with`?

Reply:
Doh!  That was just lazy programming...  Will fix.